app/workers/expire_rent_email_worker.rb

```diff
@@ -0,0 +1,8 @@
+class ExpireRentEmailWorker
+  include Sidekiq::Worker
+
+  def perform
+    rents = Rent.where(end_date: Time.zone.today)
+    rents.map { |rent| RentMailer.expire_rent_send(rent.id).deliver_now }
```

Comment:
You could pass the entire rent object to avoid made the query at the query mailer and the worker (are two queries or more if there are many rents that expire today)

Reply:
I'm not sure if that can be done  because this uses sidekiq under  the hood, and sidekiq serializes the objects, and a record  cannot be serialized correctyl

Reply:
Done

Reply:
But he is passing objects between classes methods when worker is running after he called

Reply:
I will try it

Reply:
It's works fine. I think as Jorge, because is only the call to the mail and not to the worker.

Reply:
Psd: If the mailer will sent later, I think the object doesn't work. What do you think?